src/rules/noMagicNumbersRule.ts

```diff
@@ -31,15 +41,32 @@ export class Rule extends Lint.Rules.AbstractRule {
         rationale: Lint.Utils.dedent`
             Magic numbers should be avoided as they often lack documentation, forcing
             them to be stored in variables gives them implicit documentation.`,
-        optionsDescription: "A list of allowed numbers.",
+        optionsDescription: Lint.Utils.dedent`
+            An optional config object may be provided with one or both of the following properties:
+                * \`"ignore-element-access": boolean\`
+                    * Ignores cases such as \`argv[4]\`
+                * \`"allowed-numbers": number[]\`
+                    * List of numbers the rule should ignore
+            `,
         options: {
-            type: "array",
-            items: {
-                type: "number",
+            type: "object",
+            properties: {
+                "allowed-numbers": {
+                    type: "array",
+                    items: {
+                        type: "number",
+                    },
+                    minLength: 1,
+                },
+                "allow-element-access": {
+                    type: "boolean",
+                },
             },
-            minLength: 1,
         },
-        optionExamples: [true, [true, 1, 2, 3]],
+        optionExamples: [true, [true, 1, 2, 3], [true, {
```

Comment:
this second example no longer matches the schema, which makes this PR a **breaking change**

Reply:
though i see you've preserved support for the old usage below.